### PR TITLE
[#55] Add a general documentation

### LINKED ISSUE #55: Add a general documentation
## Why

To enable new users to adopt it. Today it is easy to generate a new project, but as a new adopter, I'm quite overwhelmed by all the different services/files. It would be important to provide a high-level representation of this infrastructure. Visual models should be used too.

## Who Benefits?

New adopters of the infra-template.

### PR DESCRIPTION
**Pull request by malparty**
- Close #55

## What happened 👀

- [x] Review README.md structure
- [x] Add a diagram
- [x] Add the steps to deploy the project (starter kit)
- [x] Review variable descriptions

## Insight 📝

The main idea was for me to reduce the need to ask questions on slack when using this template for the first time! :) This way we can save time/effort for everyone. ;-)

The diagram is modified and stored in [the Infrastructure-template folder](https://lucid.app/documents#/documents?folder_id=299724278) in our Lucid account!

### Main Template README.md

image

### README.md of the generated project:

image

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,28 +1,96 @@
...
 ```
 
-* With npx:
+or
+
 ```bash
 npx @nimblehq/infra-template generate {project-name}
 ```
 
+## Deploy your infrastructure
```
Comment (andyduong1920) [inline README.md:29]: As discussed, should we move this part `Deploy your infrastructure` to the `generated project's readme`

Comment (hoangmirs) [reply README.md:29]: We can create a Documentation add-on and will copy this content into the needed folder, maybe the wiki or README of the generated project 🤔

File: README.md [hunk 2]
```diff
@@ -1,28 +1,96 @@
...
+## Starter kit
+
+The starter kit has 3 workspaces:
+- **Shared**, that handles a single ECR (Elastic Container Registry) for the whole infrastructure
+- **Staging** and **Prod**: Both workspaces are similar but use different resource names (using the suffixes `-staging` and `-prod`)
+
+### Step 1, Source Code
+
+In the source code:
+- create a `base` and a `shared` folder.
```
Comment (andyduong1920) [inline README.md:50]: I think for these steps, we could let the template does it, like generating the `shared` folder beforehand, so we could avoid some manual step like this, what do you think @malparty ? If that makes sense, we could create an issue for task :D

Comment (malparty) [reply README.md:50]: Issue created! :) 
- #85

File: README.md (hunk 1 above)
Comment (malparty) [reply README.md]: IMO we do not need a "documentation add-on", it should be placed in the `common` files when a projected variant is created (this documentation is specific to the AWS variant). 

Here is my suggested implementation: 19f1c21 ✅

File: skeleton/aws/README.md [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+
+While advanced users won't need it, the next part provides a "starter kit" architecture that suits most of the new projects.
+
+## Starter kit
+
+The starter kit has 3 workspaces:
+- **Shared**, that handles a single ECR (Elastic Container Registry) for the whole infrastructure
+- **Staging** and **Prod**: Both workspaces are similar but use different resource names (using the suffixes `-staging` and `-prod`)
+
+### Step 1, Source Code
```
Comment (hoangmirs) [inline skeleton/aws/README.md:19]: This step can be removed once we implement https://github.com/nimblehq/infrastructure-templates/issues/85, right?

**Review by hoangmirs — APPROVED** [review]
LGTM 💪

File: skeleton/aws/README.md (hunk 3 above)
Comment (malparty) [reply skeleton/aws/README.md:19]: Yes, definitely! ;-)

File: skeleton/aws/README.md [hunk 4]
```diff
@@ -0,0 +1,84 @@
+## Deploy your infrastructure
```
Comment (longnd) [inline skeleton/aws/README.md:1]: Should we move it to Wiki instead?

File: skeleton/aws/README.md [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+- create a `base` and a `shared` folder.
+- move the `main.tf`, `outputs.tf` and `variables.tf` into the `base` folder.
+- create 3 empty files (`main.tf`, `outputs.tf` and `variables.tf`) in the `shared` folder.
+- extract from the `base/*` files all data related to `ECR`.
+
+### Step 2, AWS Access Key
+
+For Terraform to push your changes into AWS, it needs an access key. There are 2 options for that:
+- Using Terraform Cloud: configure the `aws_access_key` and `aws_secret_key` as sensitive variables in your Terraform Cloud Workspaces. Then add the reference into the source code:
+  ```
```
Comment (longnd) [inline skeleton/aws/README.md:31]: to enable code syntax hightlight
```suggestion
  ```hcl
```

**Review by longnd — APPROVED** [review]
The diagram should be in SVG format :D

File: skeleton/aws/README.md (hunk 4 above)
Comment (malparty) [reply skeleton/aws/README.md:1]: The idea is to have it in the generated project. Though the generated project does not have a wiki (and I do not think we aim for that as we usually use the Wiki of the Web repository). 
So it seems good to me, that we provide this piece of info in the README file and then the user can move it where (s)he wants ;-)

File: skeleton/aws/README.md (hunk 5 above)
Comment (malparty) [reply skeleton/aws/README.md]: Nice catch!
834a87b ✅

**Comment by hoangmirs** [comment]
> The diagram should be in SVG format :D

Did you miss this comment? @malparty 
Please update it, then I'll merge this PR 🙇

**Comment by malparty** [comment]
@hoangmirs thanks for the reminder! :-) 
791ba29 ✅

File: README.md [hunk 6]
```diff
@@ -1,24 +1,25 @@
...
 
-1. Create a folder name and cd into that folder
+**Supported flavors:**
+- A `Basic` flavor is in the plan, but not available yet
+- The `Complete` flavor generates the following infrastructure:
 
-```bash
-mkdir awesome-infra && cd awesome-infra
-```
+![Diagram of the Complete Infrastructure](https://github.com/nimblehq/infrastructure-templates/blob/develop/img/diagram_complete.svg?raw=true)
```
Comment (hoangmirs) [inline README.md:11]: Can we use the relative path? @malparty 

```suggestion
![Diagram of the Complete Infrastructure](/img/diagram_complete.svg)
```

Comment (malparty) [reply README.md]: 14fc56f ✅ 🚀

File: skeleton/aws/README.md (hunk 4 above)
Comment (longnd) [reply skeleton/aws/README.md:1]: That makes sense 👍

**Pull request merged**